aws/aws-iot-device-sdk-js-v2
# Major version upgrades for typescript and typedoc

* Update typescript to latest
* Update typedoc to almost-latest MV
* Update typedoc config to account for all the breaking changes
* Swap typedoc plugins to modern ones that still work and do approximately what we had plugins doing before.
* Re-export service clients to address pickier type-checking in 4.x
* Update sample package.jsons and fix some misc. type-check errors

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
